gitlab/resource_gitlab_deploy_key_enable.go

```diff
@@ -0,0 +1,127 @@
+package gitlab
+
+import (
+	"fmt"
+	"log"
+	"strconv"
+	"strings"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	gitlab "github.com/xanzy/go-gitlab"
+)
+
+func resourceGitlabDeployEnableKey() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceGitlabDeployKeyEnableCreate,
+		Read:   resourceGitlabDeployKeyEnableRead,
+		Delete: resourceGitlabDeployKeyEnableDelete,
+		Importer: &schema.ResourceImporter{
+			State: resourceGitlabDeployKeyEnableStateImporter,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"project": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"key_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"title": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"key": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"can_push": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceGitlabDeployKeyEnableCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*gitlab.Client)
+	project := d.Get("project").(string)
+	key_id, err := strconv.Atoi(d.Get("key_id").(string))
+
+	log.Printf("[DEBUG] enable gitlab deploy key %s/%d", project, key_id)
+
+	deployKey, _, err := client.DeployKeys.EnableDeployKey(project, key_id)
+	if err != nil {
+		return err
+	}
+
+	d.SetId(fmt.Sprintf("%s:%d", project, deployKey.ID))
+
+	return resourceGitlabDeployKeyEnableRead(d, meta)
+}
+
+func resourceGitlabDeployKeyEnableRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*gitlab.Client)
+	project := d.Get("project").(string)
+	deployKeyID, err := strconv.Atoi(d.Get("key_id").(string))
+	if err != nil {
+		return err
+	}
+	log.Printf("[DEBUG] read gitlab deploy key %s/%d", project, deployKeyID)
+
+	deployKey, response, err := client.DeployKeys.GetDeployKey(project, deployKeyID)
+
+	// HTTP 404 indicates that the key is not enabled on the project, maybe it was removed through the UI
```

Comment:
We do not ignore 404 anymore. Please remove this.

Reply:
I think in this case we actually need it , see the following example workflow 

* Key is enabled  on project X through terraform , resource is not in the state so is not read at first and just created
* Resource is now in the state
* Human being disable the key by hand through the ui
* Terraform is rerun and get 404 back from gitlab

In this case unless i can control this response , and set the resource Id to "" , terraform will just error out instead of re-enabling the key 

This is my first ever terraform resource so i might just be wrong :) 

Am i missing something ?

Reply:
That is the point. If someones removes it from the UI, terraform should error, then it must me removed from state (terraform state rm).

Things should not disappear from the state magically

Reply:
Ok, quite an _opinionated_ point but i actually like it, agree with it and probably will reuse it myself.

Will fix now